pkg/proxy/iptables/proxier.go

```diff
@@ -585,6 +564,22 @@ func addHcUpdate(updateMap hcUpdateMap, name types.NamespacedName, isLocal bool)
 	}
 }
 
+func addEndpointToMap(endpointsMap map[string]endpointsInfoSlice, ip string, port api.EndpointPort, isLocal bool) {
+	// Only add the hostport if the address is valid,
+	// but keep track of port names with invalid addresses
+	// so we can clean them up later
```

Comment:
Actually no, that was the point of logging but then assigning the empty endpointsInfoSlice to the map:

`endpointsMap[port.Name] = endpoints`

even if `ip` or `port` are invalid.  We do need it later.